src/txmempool.h

```diff
@@ -667,7 +667,8 @@ class CTxMemPool
      *  for).  Note: vHashesToUpdate should be the set of transactions from the
      *  disconnected block that have been accepted back into the mempool.
      */
-    void UpdateTransactionsFromBlock(const std::vector<uint256>& vHashesToUpdate) EXCLUSIVE_LOCKS_REQUIRED(cs, cs_main) LOCKS_EXCLUDED(m_epoch);
+    void UpdateTransactionsFromBlock(const std::vector<uint256>& vHashesToUpdate,
+            uint64_t ancestor_size_limit, uint64_t ancestor_limit) EXCLUSIVE_LOCKS_REQUIRED(cs, cs_main) LOCKS_EXCLUDED(m_epoch);
```

Comment:
mmmm maybe. In general in the mempool I think the functions accept parameters with limits as opposed to being static in the mempool, and I think it would be confusing to be inconsistent. I like that passing in limits makes it explicit what the code will do, and forces any caller to consider what limit they should use.

Where these limits get enforced later they are set by reading configuration parameter at the time of transaction acceptance rather than on startup, so altering this behavior could lead to an inconsistency down the line if someone at runtime increased the setting.

Reply:
> In general in the mempool I think the functions accept parameters with limits

I think that's more a historical artifact that until #19556, the mempool was a global that was constructed before `main()`. If we were building this from scratch, I think we'd make any mempool parameters that can't be changed at runtime const and set in the ctor, rather than using globals.

> lead to an inconsistency down the line if someone at runtime increased the setting.

Changing these limits during runtime seems like a terrible idea! There would be all kinds of edge-cases where it'd be difficult to ensure that invariants aren't being invalidated.

Anyway, your choice. This is just a suggestion and could be done after this PR. Whenever I see function arguments where the same value is always passed (eg `UpdateTransactionsFromBlock()` in this PR), then it suggests to me that the interface can be simplified.

Reply:
> Changing these limits during runtime seems like a terrible idea!

Agree. Though, the tx_pool fuzz targets do this and they haven't (yet) crashed :grimacing: